web/src/main/java/org/akaza/openclinica/service/ImportServiceImpl.java

```diff
@@ -325,16 +335,30 @@ public boolean validateAndProcessDataImport(ODMContainer odmContainer, String st
                         } //itemGroupDataBean for loop
 
 
-                        if ((formDataBean.getEventCRFStatus().equals(COMPLETE) || formDataBean.getEventCRFStatus().equals(DATA_ENTRY_COMPLETE)) && itemCountInForm.getInsertedUpdatedSkippedItemCountInForm() == itemCountInForm.getItemCountInFormData()) {                         // update eventcrf status into Complete
-                            // Update Event Crf Status into Complete
-                            eventCrf = updateEventCrf(eventCrf, userAccount, Status.UNAVAILABLE,new Date());
-                            openRosaSubmissionController.updateStudyEventStatus(tenantStudy.getStudy() != null ? tenantStudy.getStudy() : tenantStudy, studySubject, studyEventDefinition, studyEvent, userAccount);
+                        if ((formDataBean.getEventCRFStatus().equals(COMPLETE) || formDataBean.getEventCRFStatus().equals(DATA_ENTRY_COMPLETE)) ) {
+                        	// update eventcrf status into Complete
+                        	if(itemCountInForm.getInsertedUpdatedSkippedItemCountInForm() == itemCountInForm.getItemCountInFormData()) {
+                        		
+                                eventCrf = updateEventCrf(eventCrf, userAccount, Status.UNAVAILABLE,new Date());
+                                openRosaSubmissionController.updateStudyEventStatus(tenantStudy.getStudy() != null ? tenantStudy.getStudy() : tenantStudy, studySubject, studyEventDefinition, studyEvent, userAccount);
 
-                            logger.debug("Form {} status updated to Complete ", formDataBean.getFormOID());
+                                logger.debug("Form {} status updated to Complete ", formDataBean.getFormOID());
+                        	}else {
+                        		// even in COMPLETE, but during import process may still get some item updated
+                        		;
+                        	}
+                            
 
                         } else if (itemCountInForm.getInsertedUpdatedItemCountInForm() > 0) {                         // update eventcrf status into data entry status
-                            // Update Event Crf Status into Initial Data Entry
-                            eventCrf = updateEventCrf(eventCrf, userAccount, Status.AVAILABLE,null);
+                           
+                        	//AC3: Complete forms with data imported into them must stay in Complete status at the conclusion of the import.
+                        	if(this.isEventCrfCompleted(eventCrf)) {
+                        		;
+                        	}else {
+                        		// Update Event Crf Status into Initial Data Entry
+                        		 eventCrf = updateEventCrf(eventCrf, userAccount, Status.AVAILABLE,null);
+                        	}
+                           
```

Comment:
I will look into the logic implementation in import for all  ACs after you review and update the code for the other comments and feedback that I added them.

Reply:
finish code changes based on your feedback, please review . thanks